browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,472 @@
+    std::unique_ptr<base::ListValue> selectors) {
+  if (!selectors || IsVettedSearchEngine(url.host())) {
+    return;
+  }
+  for (size_t i = 0; i < selectors->GetSize(); i++) {
+    base::ListValue* selectors_list = nullptr;
+    if (!selectors->GetList()[i].GetAsList(&selectors_list) ||
+        selectors_list->GetSize() == 0) {
+      continue;
+    }
+    std::string json_selectors;
+    if (!base::JSONWriter::Write(*selectors_list, &json_selectors) ||
+        json_selectors.empty()) {
+      json_selectors = "[]";
+    }
+    // Building a script for stylesheet modifications
+    std::string new_selectors_script =
+        base::StringPrintf(k_selectors_inject_script, json_selectors.c_str());
+    if (selectors_list->GetSize() != 0) {
+      auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+      if (!frame_host)
+        return;
+      frame_host->ExecuteJavaScriptInIsolatedWorld(
+          base::UTF8ToUTF16(new_selectors_script),
+          base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+    }
+  }
+
+  if (!enabled_1st_party_cf_filtering_) {
+    auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+    if (!frame_host)
+      return;
+    frame_host->ExecuteJavaScriptInIsolatedWorld(
+        base::UTF8ToUTF16(
+            *BraveCosmeticResourcesTabHelper::observing_script_),
+        base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+  }
+}
+
+void BraveCosmeticResourcesTabHelper::ProcessURL(
+    content::RenderFrameHost* render_frame_host, const GURL& url,
+    const bool& do_non_scriplets) {
+  content::CosmeticFiltersCommunicationImpl::CreateInstance(render_frame_host,
+      this);
+  if (!render_frame_host || !ShouldDoCosmeticFiltering(web_contents(), url)) {
+    return;
+  }
+  g_brave_browser_process->ad_block_service()->GetTaskRunner()->
+      PostTaskAndReplyWithResult(FROM_HERE,
+          base::BindOnce(GetUrlCosmeticResourcesOnTaskRunner,
+            url.spec()),
+          base::BindOnce(&BraveCosmeticResourcesTabHelper::
+            GetUrlCosmeticResourcesOnUI, AsWeakPtr(),
+            content::GlobalFrameRoutingId(
+                  render_frame_host->GetProcess()->GetID(),
+                  render_frame_host->GetRoutingID()),
+            url.spec(), do_non_scriplets));
+}
+
+void BraveCosmeticResourcesTabHelper::DidFinishNavigation(
+    content::NavigationHandle* navigation_handle) {
+  if (!navigation_handle)
+    return;
+  ProcessURL(navigation_handle->GetRenderFrameHost(),
+      web_contents()->GetLastCommittedURL(),
+      navigation_handle->IsInMainFrame());
+}
+
+void BraveCosmeticResourcesTabHelper::ResourceLoadComplete(
+    content::RenderFrameHost* render_frame_host,
+    const content::GlobalRequestID& request_id,
+    const blink::mojom::ResourceLoadInfo& resource_load_info) {
+  ProcessURL(render_frame_host, resource_load_info.final_url, false);
+}
+
+void BraveCosmeticResourcesTabHelper::HiddenClassIdSelectors(
```

Comment:
thismethod  name should really be a verb

Reply:
corrected to `ApplyHiddenClassIdSelectors`